src/d3a/models/strategy/area_agents/one_sided_alternative_pricing_agent.py

```diff
@@ -39,7 +38,7 @@ def _get_children_by_name(area, name):
         return next((c for c in area.children if c.name == name), None)
 
     def _buy_energy_alternative_pricing_schemes(self, area):
-        if not _is_house_node(self.owner):
+        if not all(child.children == [] for child in self.owner.children):
```

Comment:
How about adding a `is_house_node` into the utily.py? You always strike me as you prefer a function than a logic  chain. :-)

Reply:
Indeed I prefer a function than a logic chain to avoid code repetition but that was the only usage of `_is_house_node `.